packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -38,6 +39,74 @@ persistCache({
     storage: window.localStorage
 });
 
+const GIFT_OPTIONS_QUERY = gql`
+    query GiftOptions {
+        gift_options {
+            include_gift_receipt
+            include_printed_card
+            gift_message
+        }
+    }
+`;
+
+const resolvers = {
+    Query: {
+        gift_options: (_, { cart_id }, { cache }) => {
+            /**
+             * This is how the `cacheKeyFromType` saves the
+             * cart data in the `InMemoryCache`.
+             */
+            const cartIdInCache = `Cart:${cart_id}`;
+
+            const {
+                include_gift_receipt,
+                include_printed_card,
+                gift_message
+            } = cache.data.data[cartIdInCache];
+
+            return {
+                __typename: 'Cart',
+                include_gift_receipt,
+                include_printed_card,
+                gift_message
+            };
+        }
+    },
+    Mutation: {
+        set_gift_options: (
+            _,
+            {
+                cart_id,
+                include_gift_receipt = false,
+                include_printed_card = false,
+                gift_message = ''
+            },
+            { cache }
+        ) => {
+            cache.writeQuery({
+                query: GIFT_OPTIONS_QUERY,
+                data: {
+                    gift_options: {
+                        include_gift_receipt,
+                        include_printed_card,
+                        gift_message,
+                        id: cart_id,
+                        __typename: 'Cart'
+                    }
+                }
+            });
+
+            /**
+             * We do not return anything on purpose.
```

Comment:
This is a good call-out. Since the component renders based on `data` changing if we start returning data we will cause a weird UX. 👍